### ISSUE TITLE
Shortcuts and Shell Command Exit Codes

### ISSUE THREAD

**Issue by nano9g** [holzschu/a-shell#226]
Currently when I try to run a shell command from Shortcuts using a-Shell, it throws an error if the command's return code is nonzero. Would it be possible to alter this behaviour so that commands such as ping can be used? [Here's a simple shortcut to demonstrate the issue](https://www.icloud.com/shortcuts/73814786c9604029a09f06d974cbbfb2) - the shortcut stops because ping returns 2 and there's no way around this I can find.

What could be really helpful is to have options that control what the Execute Command action returns - command output, return code, or maybe even both as a dictionary.

Thank you!

(I've attempted to work around this by using Python, Lua, Perl, and wasm but they all failed in different ways… I can provide details if needed.)

**Comment by holzschu** [comment]
Hi, 
Interesting question. I'm going to have a look.

**Comment by holzschu** [comment]
I see where the issue is. It's only an issue when running commands "In extension", right?
Making sure the shortcut doesn't stop after the command returns something other than 0 is easy. I'll add a settings for that, with the default value being the current behaviour.
Selecting what to return is more complicated; you can already access the command output by redirecting to a file, then reading that file.

**Comment by holzschu** [comment]
You could also work around the issue by calling: `sh ping -c 1 -W 1 169.0.0.0 && true`, but I will add the settings anyway.

`sh command1 && command2 && ...` will execute commands in sequence until one of them returns 0.
`sh command1 || command2 || ...` will execute commands in sequence until one of them returns non-zero.

**Comment by nano9g** [comment]
Thanks for the quick response and the workaround! I hadn't been able to determine the proper syntax to suppress the return code with a-Shell sh, so that's very useful. And of course, options in the UI will make it simpler for the future.

**Comment by holzschu** [comment]
Hi, 
the new TestFlight version is out now, and it has the option you suggested: https://testflight.apple.com/join/WUdKe3f4

**Comment by nano9g** [comment]
Works perfectly, thank you!

**Comment by yur1xpp** [comment]
Hey @holzschu I have the command `ping -c 1 -W 1 8.8.8.8 > /dev/null && echo 1 || echo 0` and run them in extensions, but I always get `0` as output together with `dash: 0: Can't open ping`. Is there something changed ever since this issue open? I'm using the official version from AppStore.

**Comment by holzschu** [comment]
Yes, since this issue was opened, we added a full shell (dash), which is aliased to sh (as it is on most platforms). So `sh` (alone) calls `dash`.

You'll need: `sh -c ping -c 1 -W 1 8.8.8.8 > /dev/null && echo 1 || echo 0` to reproduce the expected behaviour.

**Comment by yur1xpp** [comment]
Thanks for the info! Seems like they don't behave like the "normal" `sh`? For example the exact command: `sh -c "ping -c 1 -W 1 8.8.8.8 > /dev/null && echo 1 || echo 0"` when running in a-Shell will always return `0`, while in linux we get `1`. Any idea?

**Comment by holzschu** [comment]
I would say that, for that expression, it depends on how the parser interprets the different operators, and in which order, a bit like "6 / 2 * 3". I'll have a look to see if it can be improved.

**Comment by yur1xpp** [comment]
Huh, got it, appreciate your effort on this!
